src/index.ts

```diff
@@ -442,7 +440,7 @@ export class AzureOpenAI extends OpenAI {
       if (!Core.isObj(options.body)) {
         throw new Error('Expected request body to be an object');
       }
-      const model = options.body['model'];
+      const model = this._deployment || options.body['model'];
```

Comment:
I think this is the right order of preference. I'm still calling it out here because I would normally say that parameters to point of invocation of client method overrides parameters to the client constructor and I'd like for others to have a chance to chime in.

Reply:
Yup I agree it feels weird. I think the main motivation behind this convoluted order is that request-level `model` is not always accessible, e.g. in non-json requests. I imagine customers will be hella confused if they supply a `model` in the request and an error comes back.